app/services/sign_in/token_response_generator.rb

```diff
@@ -74,5 +77,15 @@ def serialized_service_account_token(access_token:)
         }
       }
     end
+
+    def create_user_audit_log(user_verification:)
+      user_action_event = UserActionEvent.find_by(identifier: 'user_login')
+      UserAuditLogger.new(user_action_event:,
+                          acting_user_verification: user_verification,
+                          subject_user_verification: user_verification,
+                          status: :success,
+                          acting_ip_address: cookies.request.remote_ip,
```

Comment:
What options do we have with this? Does it have to come off the `cookies` object? If this class has access to it, does that mean we can hide this inside `UserAuditLogger` class instead? Do we need to make any considerations if the 'cookie' object does not exist?

Reply:
> Does it have to come off the `cookies` object?

No, we could pull directly from the Rails `request` object that's accessible at the controller level. That would make calling it more difficult in some places.

> If this class has access to it, does that mean we can hide this inside UserAuditLogger class instead?

`TokenResponseGenerator` has access to the cookies because they were already being passed in directly by the calling `SignInController`.

> Do we need to make any considerations if the 'cookie' object does not exist?

Refactoring to make all `UserAuditLogger` calls come either directly from a controller or from a method that has had the `request` object values passed to it should prevent the need to interact with cookies for this.